Automattic/wp-calypso
# Odyssey: Add opt-out notice for odyssey

#### Proposed Changes

* Adds`packages/components/src/notice-banner` which is forked from [Notice](https://github.com/Automattic/jetpack/tree/trunk/projects/js-packages/components/components/notice) adjusted to use Calypso standard
* Adds Opt-out notice to Traffic page highlight section
* Adds supports to access existing `jetpack/v4` API directly
* Sends dismissing notice request to JITM endpoint for the site to remember the notice is dismissed

#### Testing Instructions

- Use the branch from https://github.com/Automattic/jetpack/pull/28733 for Jetpack the plugin
- Setup Odyssey with Option 1 in `PejTkB-3E-p2` <- enable New Stats if not yet enabled
- Open `/wp-admin/admin.php?page=stats`
- Ensure an opt-out notice is shown on the Traffic page
- Click `manage your settings`
- Ensure you are taken to Jetpack Stats settings section
- Return to `/wp-admin/admin.php?page=stats`
- Click `x`
- Ensure the notice is dismissed
- Refresh the page
- Ensure the notice wouldn't appear again


<img width="1291" alt="image" src="https://user-images.githubusercontent.com/1425433/216485797-e9b34545-427a-46ea-a342-3ba4a7cdebc4.png">


#### Pre-merge Checklist





Related to #72736
